### PR TITLE
Add a CONTRIBUTING.md file

### LINKED ISSUE #416: CONTRIBUTING.md
Would be nice to have a markdown file that explains contribution process. Could involve coding style, things to consider and especially which branch to base on & send PRs to.

### PR DESCRIPTION
**Pull request by astorije**
See https://github.com/blog/1184-contributing-guidelines to know why :-)

Supersedes #358, closes #416 and #478.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+the history. The general rules are to [use the imperative present
+tense](https://git-scm.com/book/ch5-2.html#Commit-Guidelines), to start with a
+single concise line, followed by a blank line and a more detailed explanation
+when necessary. Tim Pope wrote an [excellent
+article](http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html)
+on how one should format their commit messages.
+
+When you send your PR, expect two different reviews from the project
+maintainers. They will make comments and give their respective :+1: before your
+changes can go into production.
```
Comment (JocelynDelalande) [inline CONTRIBUTING.md]: Maybe be a bit more explicit, the `:+1` notion is not universal.

**Comment by JocelynDelalande** [comment]
Maybe include a list of maintainers somewhere, so that people know in last restort where to target harassment if something is inactive for too long.

File: CONTRIBUTING.md (hunk 1 above)
Comment (astorije) [inline CONTRIBUTING.md]: Addressed in 1671858.

**Comment by astorije** [comment]
@JocelynDelalande, I agree. A couple of solutions though:

1. Naively list the contributors in that CONTRIBUTING.md file
2. Add a [`contributors` entry to the `package.json` file](https://docs.npmjs.com/files/package.json#people-fields-author-contributors).

I believe there is also a `maintainers` field but I could not find information about it.

So, what's your call?

**Comment by dideler** [comment]
Looks very good!

Re: list of maintainers/contributors, is https://github.com/erming/shout/graphs/contributors sufficient?

**Comment by JocelynDelalande** [comment]
>     Naively list the contributors in that CONTRIBUTING.md file

The thing I was speaking about is listing Maintainers, not contributors.

>     Add a contributors entry to the package.json file.

that's for credit, not contributing guidelines

> I believe there is also a maintainers field but I could not find information about it.

Can be filled, but that's a npm-package level information, not a "how to contribute nicely" information. I'd prefer the information to be as close as possible to contribution guidelines.

I'm not against crediting contributors explicitly somewhere (git log implicitly credits them), I even find it nice, but that's another discussion.

**Comment by astorije** [comment]
Well, my bad for not being explicit enough. I understood your initial
comment and was looking into ways to list ourselves. I was not planning to
list contributors.

My suggestion is to list the maintainers in the maintainers field of the
package.json file and to add a link in the CONTRIBUTING to this list. One
place to maintain and still fairly human-readable. That said, if you really
prefer listing the maintainers in the CONTRIBUTING, I won't object.

**Comment by JocelynDelalande** [comment]
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA1

> My suggestion is to list the maintainers in the maintainers
> field of the package.json file and to add a link in the
> CONTRIBUTING to this list. One place to maintain and still
> fairly human-readable. 

ok, got it, anyway "maintainers" seems to be [made to receive
npm-account
informations](https://docs.npmjs.com/files/package.json#people-fields-author-contributors)
(the doc is a bit unclear though...).

> That said, if you really prefer listing
> the maintainers in the CONTRIBUTING, I won't object.

I'd prefer that, it's simpler, *packages.json* is not a document
for humans. Require a bit of manual sync though, but no drama if
it's sometimes out of sync...

-----BEGIN PGP SIGNATURE-----
Version: GnuPG v1.4.12 (GNU/Linux)

iQIcBAEBAgAGBQJWFoS3AAoJEOodh6AvlLNTaYsP/2DWnhDwJE24EnAAUYhM/YuN
puPTbOHTLT8eZnhNAcg2uYpiD/6bI08I4tu0ruDz4emSvB8NEJdiEzJ+jvkEKbyN
FV4sycLvYm3YcPWv2jHa3JuXKxbg1ZcrKfWbW7flhEtjgVf5GmnW5fhZcyUw+noS
21HVXcgamOclJacH/tGAVthRISFm1W9LKrwXprO7sfZhTrZUwSxz9WM4K+zLzDlW
NWGQSX31nOoWymH7JLcBBbsZRAmbMgR9Zwb4vHMPFa0gTM4SLhUrtRSN15q5WAb6
zG20f0S3cI7nXUcANkipCVzAY2Zi6JiBbzW3T7hnlxkvHlmsMcrNF+Own1bJLUMT
smowtV5hvdUVQEoGjuaTMjonuEZ6BoxngZH3VFwxmCJpipVCG10r+2rdnKEp++yJ
41c5NVz8aCf/AEuDe1SdwtCSN9WOebpcL8PKg268PaGIDqHqX5xHmBtWxvRa+gk6
2MaX45l182Uq4eBmdFxQr3SsLN2dqSgzFGYxA0yU8/o9gjWIaBvNKI/uBUpOF3z5
t3U08E4DDnlNNaJwGMh/2kT6MrVRRze8Ko3FcgX7ngVsPP8mCaXdUuXlruhi0WMt
XOLeg2jI0TQ/Cp6HS0xoqUbA4MVksdmn9CsS75N32V+i/D2pZah7oOiYM7gvtVkm
IrzcL4ZkSLn+odXNeqq4
=vzNe
-----END PGP SIGNATURE-----

**Comment by astorije** [comment]
Fair enough, will do.

**Comment by astorije** [comment]
Added in 9f7f0f7.

**Comment by JocelynDelalande** [comment]
:+1:

**Pull request merged**